Content.Server/_RMC14/Marines/Roles/Ranks/RankSystem.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Content.Server.Chat.Systems;
+using Content.Server.GameTicking;
+using Content.Shared._RMC14.Marines.Roles.Ranks;
+using Content.Shared.Access.Systems;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server._RMC14.Marines.Roles.Ranks;
+
+public sealed class RankSystem : SharedRankSystem
+{
+    [Dependency] private readonly SharedIdCardSystem _idCardSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototypes = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<RankComponent, TransformSpeakerNameEvent>(OnSpeakerNameTransform);
+        SubscribeLocalEvent<PlayerSpawnCompleteEvent>(OnPlayerSpawnComplete);
+    }
+
+    private void OnSpeakerNameTransform(EntityUid uid, RankComponent component, TransformSpeakerNameEvent args)
+    {
+        var name = GetSpeakerRankName(uid);
+        if (name == null)
+            return;
+
+        args.Name = name;
+    }
+
+    private void OnPlayerSpawnComplete(PlayerSpawnCompleteEvent ev)
+    {
+        var uid = ev.Mob;
+        var jobId = ev.JobId;
+
+        if (jobId == null)
+            return;
+
+        if (_idCardSystem.TryFindIdCard(uid, out var idcard))
+        {
+            var idCardEntity = idcard.Owner;
+
+            if (jobId == null)
+                return;
```

Comment:
Redundant check if I'm not wrong, since this is done a few lines up

Reply:
thanks